plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java

```diff
@@ -139,12 +140,18 @@ public ConnectorPageSource createPageSource(
                 .collect(toImmutableList());
 
         Map<String, Optional<String>> partitionKeys = split.getPartitionKeys();
-
+        ColumnMappingMode columnMappingMode = getColumnMappingMode(table.getMetadataEntry());
         Optional<List<String>> partitionValues = Optional.empty();
         if (deltaLakeColumns.stream().anyMatch(column -> column.getBaseColumnName().equals(ROW_ID_COLUMN_NAME))) {
             partitionValues = Optional.of(new ArrayList<>());
             for (DeltaLakeColumnMetadata column : extractSchema(table.getMetadataEntry(), typeManager)) {
-                Optional<String> value = partitionKeys.get(column.getName());
```

Comment:
This looks like an existing bug. Did the name mismatching affect final results? e.g. correctness issue

Reply:
Yes it did, because of that TableChangesSplitSource didn't have correct names/values of partition columns